tests/test_code_actions.py

```diff
@@ -257,6 +257,30 @@ def test_requests_with_no_diagnostics(self) -> Generator:
         self.assertEquals(annotations_range[0].a, 3)
         self.assertEquals(annotations_range[0].b, 0)
 
+    def test_extends_range_to_include_diagnostics(self) -> Generator:
+        def handle_response(actions_by_config: CodeActionsByConfigName) -> None:
+            pass
+        self.insert_characters('x diagnostic')
+        yield from self.await_message("textDocument/didChange")
+        yield from self.await_client_notification(
+            "textDocument/publishDiagnostics",
+            create_test_diagnostics([
+                ('diagnostic word', Range(Point(0, 2), Point(0, 12))),
+                ('all content', Range(Point(0, 0), Point(0, 12))),
+            ])
+        )
+        self.view.sel().clear()
+        self.view.sel().add(sublime.Region(0, 5))
+        self._trigger_on_selection_modified_async()
+        yield 100
```

Comment:
Why do we need these `yield 100`?

Reply:
`on_modified_async` is async and also it triggers code actions asynchronously (from `debounced`) so we need to delay things a bit to wait for `on_modified_async` especially as it seems to be a bit random when it's coming.

It might cause randomness in tests but seems pretty stable for me with 100ms.

Reply:
But the `self.await_message('textDocument/codeAction')` already takes care of waiting?

Reply:
But it will take previous response if there was one. And there is one triggered implicitly before on adding characters.

Reply:
Hmm, I see.